### ISSUE TITLE
Reactive Rest Client + CloudEvents shenanigans

### ISSUE THREAD

**Issue by nitroin** [quarkusio/quarkus#30596]
### Describe the bug

I'm opening this issue in correlation with the sibling [one here](https://github.com/cloudevents/sdk-java/issues/465) on the cloudevents repo.

Apparently using the reactive rest client in conjunction with cloudevents inhibits the correct headers and content negotiation when trying to dispatch "structured encoded" events with the following annotation:

```java
@POST
@Consumes(JsonFormat.CONTENT_TYPE)
Response publishCloudEvent(@PathParam("channel") String channel, CloudEvent cloudEvent);
```

I'm opening this issue because using `quarkus-rest-client-jackson` instead of the suggested `quarkus-rest-client-reactive-jackson` results in a correct behavior.

### Expected behavior

The rest client dispatch "structured encoded" cloudevents. See [here](https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/http-protocol-binding.md#32-structured-content-mode) for the detailed spec.

Eg, an HTTP POST with the following body payload:

```json
{
    "specversion" : "1.0",
    "type" : "User",
    "source": "io.cloudevents.examples/user",
    "id": "536808d3-88be-4077-9d7a-a3f162705f78",
    "subject": "SUBJ-0001",
    "data" : {
        "username": "jsmith",
        "firstName": "John",
        "lastName": "Smith",
        "age": 37
    }
}
```

### Actual behavior

The rest client always dispatch "binary encoded" cloudevents. Ignoring the `@Consumes` annotation. See [here](https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/http-protocol-binding.md#31-binary-content-mode) for the detailed spec.

Eg. an HTTP POST with the following body payload:

```json
{
  "username": "jsmith",
  "firstName": "John",
  "lastName": "Smith",
  "age": 37
}
```

### How to Reproduce?

See [this](https://github.com/nitroin/cloudevents-quarkus-structured-repro) repo.

TL;DR:

- Checkout https://github.com/nitroin/cloudevents-quarkus-structured-repro
- Configure a server to receive the produced cloudevent in `application.properties` 
- Start quarkus with `mvn quarkus dev` or `quarkus dev`
- Navigate to http://localhost:8080/hello
- Inspect the dispatched cloudevents

### Output of `uname -a` or `ver`

Linux kunta-dell 5.15.79.1-microsoft-standard-WSL2 #1 SMP Wed Nov 23 01:01:46 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17" 2021-09-14 OpenJDK Runtime Environment Microsoft-26987 (build 17+35) OpenJDK 64-Bit Server VM Microsoft-26987 (build 17+35, mixed mode, sharing)

### Quarkus version or git rev

2.15.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

**Comment by nitroin** [comment]
I'm closing this issue after verifying that removing a third-party dependency results in a correct behavior. See [here](https://github.com/cloudevents/sdk-java/issues/465#issuecomment-1403410774).

**Comment by geoand** [comment]
Thanks for the update
